base/shared/src/main/scala/scalaz/tc/foldable.scala

```diff
@@ -6,13 +6,24 @@ import scala.List
 import scala.language.experimental.macros
 
 trait FoldableClass[F[_]] {
+  /**
+  Applies a binary operator to a start value and all elements of this sequence, going left to right.
+  Note: this will not terminate for infinite-sized collections. For Example, Stream.
+  */
   def foldLeft[A, B](fa: F[A], z: B)(f: (B, A) => B): B
-
+  /**
+  Applies a binary operator to all elements of this list and a start value, going right to left.
+  */
   def foldRight[A, B](fa: F[A], z: => B)(f: (A, => B) => B): B
-
+  /**
+  It traverse the collection type A and does transform to B by using F function.
+  */
```

Comment:
This one needs to be expanded. It makes no mention of the fact that the translation `a -> b` where `Monoid b => b`, allows the function to collapse the functor using `mappend`.